### PR TITLE
Fix for config problem

### PR DESCRIPTION
**Pull request by KillerOfPie**
[contributing]: https://github.com/SparklingComet/TradeShop/blob/master/.github/CONTRIBUTING.md

## Pull Request Etiquette

There are several guidelines you should follow in order for your
  Pull Request to be merged.

- [:heavy_check_mark:] I have checked the PRs for upcoming features/bug fixes.
- [:heavy_check_mark:] I have read the [contributing guidelines][contributing].

> It is sometimes better to include more changes in a single commit. 
  If you find yourself having an overwhelming amount of commits, you
  can **rebase** your branch.

### Pull Request

Fix for issues #46 & #43

### CODE REVIEW

File: src/main/java/org/shanerx/tradeshop/TradeShop.java [hunk 1]
```diff
@@ -49,364 +49,377 @@
...
+
+	private void addIllegalItems() {
+		blacklist.clear();
+		blacklist.add("air");
+		for (String s : getSettings().getStringList("illegal-items")) {
+			blacklist.add(s.toLowerCase());
+		}
+	}
+
+	private int addMessage(String node, String message) {
```
Comment (KillerOfPie) [inline src/main/java/org/shanerx/tradeshop/TradeShop.java:209]: Change Here

File: src/main/java/org/shanerx/tradeshop/TradeShop.java [hunk 2]
```diff
@@ -49,364 +49,377 @@
...
+		if (messages.get(node) == null) {
+			messages.set(node, message);
+			return 1;
+		}
+
+		return 0;
+	}
+
+	private int addSetting(String node, Object value) {
```
Comment (KillerOfPie) [inline src/main/java/org/shanerx/tradeshop/TradeShop.java:218]: Change Here

File: src/main/java/org/shanerx/tradeshop/TradeShop.java [hunk 3]
```diff
@@ -49,364 +49,377 @@
...
+		if (settings.get(node) == null) {
+			settings.set(node, value);
+			return 1;
+		}
+
+		return 0;
+	}
+
+	private void addMessageDefaults() {
```
Comment (KillerOfPie) [inline src/main/java/org/shanerx/tradeshop/TradeShop.java:227]: Change Here

File: src/main/java/org/shanerx/tradeshop/TradeShop.java [hunk 4]
```diff
@@ -49,364 +49,377 @@
...
+		toSave += addMessage("missing-item", "&cYour sign is missing an item for trade.");
+		toSave += addMessage("missing-info", "&cYour sign is missing necessary information.");
+		toSave += addMessage("amount-not-num", "&cYou should have an amount before each item.");
+		toSave += addMessage("buy-failed-sign", "&cThis shop sign does not seem to be formatted correctly, please notify the owner.");
+
+		if (toSave > 0)
+			save();
+	}
+
+	private void addSettingsDefaults() {
```
Comment (KillerOfPie) [inline src/main/java/org/shanerx/tradeshop/TradeShop.java:281]: Change Here

File: src/main/java/org/shanerx/tradeshop/TradeShop.java [hunk 5]
```diff
@@ -49,364 +49,377 @@
...
+			}
+
+		if (customItems != null)
+			try {
+				customItems.save(customItemsFile);
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+
+		reloadConfig();
```
Comment (KillerOfPie) [inline src/main/java/org/shanerx/tradeshop/TradeShop.java:402]: Change Here

**Review by KillerOfPie — COMMENTED** [review]
Marking where changes were made

File: src/main/java/org/shanerx/tradeshop/TradeShop.java (hunk 1 above)
Comment (KillerOfPie) [reply src/main/java/org/shanerx/tradeshop/TradeShop.java:209]: Returns 1 if changed and 0 if not

File: src/main/java/org/shanerx/tradeshop/TradeShop.java (hunk 2 above)
Comment (KillerOfPie) [reply src/main/java/org/shanerx/tradeshop/TradeShop.java:218]: Returns 1 if changed and 0 if not

File: src/main/java/org/shanerx/tradeshop/TradeShop.java (hunk 3 above)
Comment (KillerOfPie) [reply src/main/java/org/shanerx/tradeshop/TradeShop.java:227]: takes all the returns from the adds and saves only if something was changed(value greater then 0)

File: src/main/java/org/shanerx/tradeshop/TradeShop.java (hunk 4 above)
Comment (KillerOfPie) [reply src/main/java/org/shanerx/tradeshop/TradeShop.java:281]: takes all the returns from the adds and saves only if something was changed(value greater then 0)

File: src/main/java/org/shanerx/tradeshop/TradeShop.java (hunk 5 above)
Comment (KillerOfPie) [reply src/main/java/org/shanerx/tradeshop/TradeShop.java:402]: Reloads the config which will re-run the add defaults however shouldn't save a second time unless settings were saved

**Review by SparklingComet — APPROVED** [review]

**Pull request merged**